lib/rpcenc/reader_test.go

```diff
@@ -31,6 +36,24 @@ func (h *ReaderHandler) ReadAllApi(ctx context.Context, r io.Reader, mustRedir b
 	return h.readApi(ctx, r)
 }
 
+func (h *ReaderHandler) ReadAllWaiting(ctx context.Context, r io.Reader, mustRedir bool) ([]byte, error) {
+	if mustRedir {
+		if err := r.(*RpcReader).MustRedirect(); err != nil {
+			return nil, err
+		}
+	}
+
+	h.cont <- struct{}{}
+	<-h.cont
+
+	var m []byte
+	m, h.subErr = h.readApi(ctx, r)
+
+	h.cont <- struct{}{}
+
+	return m, h.subErr
+}
```

Comment:
This method seems to only be for testing, should it not be a public method on `ReadHandler`?

Reply:
`ReadHandler` is also private to the test. The method must be public for go-jsonrpc to pick it up when creating test rpc server.

Reply:
Oh dump me, it's defined in this file.